site/content/api/Metrics.md

```diff
@@ -56,23 +56,26 @@ The names of Hono's standard components are as follows:
 
 Additional tags for protocol adapters are:
 
-| Tag        | Value                              | Description |
-| ---------- | ---------------------------------- | ----------- |
-| *tenant*   | *string*                           | The name of the tenant that the metric is being reported for |
-| *type*     | `telemetry`, `event`             | The type of message that the metric is being reported for |
+| Tag        | Value                                                  | Description |
+| ---------- | ------------------------------------------------------ | ----------- |
+| *qos*      | [ `N/A`, `0`, `1` ]                                   | The quality of service level used for a message. |
```

Comment:
Does it make sense to add `N/A` as a value to this list? I would assume that if there is no QoS, then the tag would not be added.

Reply:
Not sure if this has any (negative) impact on e.g. query performance in Prometheus. I will try to find out ...

Reply:
Prometheus has no problem with some time series instance vector not having a specific tag set while other instance vectors of the same metric name have it set. However, the Micrometer API does not allow to pass in a `null` value for a tag. Thus, it would require extra handling of unknown QoS value in the metrics code. IMHO there should be only very rare cases where a QoS value is unknown, so I'd rather keep the `N/A` value.

Reply:
So I think this could very easily be solved by adding an extra method to the enum, and letting it handle with the different cases.

Butting adding some arbitrary value (`N/A`) as a replacement of `null` sounds like a bad idea. I think not adding a value in the case there is none, is the right approach.

Reply:
> So I think this could very easily be solved by adding an extra method to the enum, and letting it handle with the different cases.

Not sure what you mean by this. Something like `QoS.asTag()` which returns a `Tag` instance or `null`?

Reply:
Either that or something like `QoS.applyTo(…)` if you don't want to add Micrometer APIs to the enum.

Reply:
What would such a `QoS.applyTo()` method do and what input parameters would it have?